hudi-common/src/main/java/org/apache/hudi/common/config/HoodieReaderConfig.java

```diff
@@ -72,4 +72,10 @@ public class HoodieReaderConfig extends HoodieConfig {
       .markAdvanced()
       .sinceVersion("1.0.0")
       .withDocumentation("Whether to use positions in the block header for data blocks containing updates and delete blocks for merging.");
+
+  public static final ConfigProperty<Boolean> ENABLE_INTERNAL_SCHEMA_CACHE = ConfigProperty
+          .key("hoodie.schema.cache.enable")
+          .defaultValue(false)
```

Comment:
Should we always cache the schema though, I mean if we have some eviction strategy for the cache, the cost would be tiny.

Reply:
Do you want me to change the defaultValue of this config to `true`, or to remove this config property completely and use internal schema cache everywhere unconditionally?

Reply:
@jonvex (as the reporter of this task) and @xiarixiaoyao (as the contributor of InternalSchemaCache): Could you please join this conversation. What do you thing about  Danny's proposal?

Reply:
> Do you want me to change the defaultValue of this config to true, or to remove this config property completely and use internal schema cache everywhere unconditionally

I'm inclined we remove the option, and do the cache where ever if possible.